kiota_http/middleware/redirect_handler.py

```diff
@@ -186,10 +186,7 @@ def _same_origin(self, url: httpx.URL, other: httpx.URL) -> bool:
         """
         Return 'True' if the given URLs share the same origin.
         """
-        return (
-            url.scheme == other.scheme and url.host == other.host
-            and self.port_or_default(url) == self.port_or_default(other)
-        )
+        return (url.scheme == other.scheme and url.host == other.host)
```

Comment:
technically a redirection from http to https would constitute an upgrade and would be fine.

Reply:
Yes, it is. This check is specifically for redirection to same scheme and host. The other method `is_https_redirect` checks redirection from `http` to `https`. Both are valid.